src/routes/api-routes.js

```diff
@@ -9,7 +10,63 @@ const getXibit = (req, res) => {
   res.send('Get 1 xibit');
 };
 
+const getAllPlans = async (_, res) => {
+  try {
+    const data = await db.Plan.find({});
+    res.status(200).json(data);
+  } catch (error) {
+    res.status(500).json(error.message);
+  }
+};
+
+const getPlanById = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const data = await db.Plan.findById(id);
+    res.status(200).json(data);
+  } catch (error) {
+    res.status(500).json(error.message);
+  }
+};
+
+const addPlans = async (req, res) => {
+  try {
+    const content = req.body;
+    const data = await db.Plan.create(content);
+    res.status(204).json(data);
+  } catch (error) {
+    res.status(500).send(error.message);
+  }
+};
+
+const updatePlan = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const content = req.body;
+    const data = await db.Plan.findOneAndUpdate(id, content, { upsert: true });
+    res.status(202).json(data);
+  } catch (error) {
+    res.status(500).send(error.message);
+  }
+};
+
+const deletePlans = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const data = await db.Plan.findByIdAndDelete(id);
+    res.status(202).json(data);
```

Comment:
Probably worth returning 200 here as 202 is 'Accepted' rather than an 'operation' success